src/react-components/styles/theme.js

```diff
@@ -9,9 +9,17 @@ if (!config && process.env.STORYBOOK_APP_CONFIG) {
   config = JSON.parse(process.env.STORYBOOK_APP_CONFIG);
 }
 
+if (!config) {
+  config = window.APP_CONFIG;
+}
+
+if (config && config.theme && config.theme.failed_to_load) {
+  console.error("Custom themes failed to load. If you are an admin, reconfigure your themes in the admin panel.");
+}
+
 export const defaultTheme = "default";
 
-export const themes = (config && config.themes) || [];
+export const themes = (config && config.theme && config.theme.themes) || [];
```

Comment:
```suggestion
export const themes = (config?.theme?.themes) || [];
```

If it fails on any of those, it will return `undefined`